File: v2/services/OrderService/src/data/infrastructure/database/models/Order.js
const {
  ORDER_STATUSES,
  PENDING_ORDER_STATUS,
} = require('../../../../common/constants');

module.exports = (sequelize, DataTypes) => {
  const Order = sequelize.define('orders', {
    orderNo: {
      type: DataTypes.UUID,
      defaultValue: DataTypes.UUIDV4,
      allowNull: false,
      unique: true,
    },
    userId: {
      type: DataTypes.UUID,
      allowNull: false,
    },
    status: {
      type: DataTypes.ENUM,
      allowNull: false,
      defaultValue: PENDING_ORDER_STATUS,
      values: ORDER_STATUSES,
    },
    amount: {
      type: DataTypes.STRING,
      allowNull: false,
    },
    currency: {
      type: DataTypes.STRING,
      allowNull: false,
    },
    productId: {
      type: DataTypes.STRING,
      allowNull: false,
    },
    quantity: {
      type: DataTypes.INTEGER,
      allowNull: false,
    },
  }, {
    freezeTableName: true,
  });
  return Order;
};

File: v1/services/WarehouseService/src/main/kotlin/com/arconsis/domain/orders/Order.kt
package com.arconsis.domain.orders

import com.arconsis.domain.shipments.Shipment
import com.arconsis.domain.shipments.ShipmentStatus
import io.quarkus.kafka.client.serialization.ObjectMapperDeserializer
import java.util.*

data class Order(
    val id: UUID,
    val userId: UUID,
    val amount: Double,
    val currency: String,
    val productId: String,
    val quantity: Int,
    val status: OrderStatus,
)

enum class OrderStatus {
    REQUESTED,
    VALIDATED,
    OUT_OF_STOCK,
    PAID,
    SHIPPED,
    COMPLETED,
    PAYMENT_FAILED,
    CANCELLED,
    REFUNDED,
    SHIPMENT_FAILED
}

fun Order.toFailedShipment(shipmentId: UUID?) = Shipment(
    id = shipmentId,
    orderId = id,
    userId = userId,
    status = ShipmentStatus.FAILED
)

class OrdersDeserializer : ObjectMapperDeserializer<Order>(Order::class.java)